tests/integration/golang/mlflow/experiment/search_test.go

```diff
@@ -1,3 +1,262 @@
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/hetiansu5/urlquery"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/api"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/api/request"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/api/response"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/dao/models"
+	"github.com/G-Research/fasttrackml/tests/integration/golang/fixtures"
+	"github.com/G-Research/fasttrackml/tests/integration/golang/helpers"
+)
+
+type SearchExperimentsTestSuite struct {
+	suite.Suite
+	client   *helpers.HttpClient
+	fixtures *fixtures.ExperimentFixtures
+}
+
+func TestSearchExperimentsTestSuite(t *testing.T) {
+	suite.Run(t, new(SearchExperimentsTestSuite))
+}
+
+func (s *SearchExperimentsTestSuite) SetupTest() {
+	s.client = helpers.NewHttpClient(os.Getenv("SERVICE_BASE_URL"))
+	fixtures, err := fixtures.NewExperimentFixtures(os.Getenv("DATABASE_DSN"))
+	assert.Nil(s.T(), err)
+	s.fixtures = fixtures
+}
+
+func (s *SearchExperimentsTestSuite) Test_Ok() {
+	// 1. prepare database with test data.
+	experiments := []models.Experiment{
+		{
+			Name: "Test Experiment 1",
+			Tags: []models.ExperimentTag{
+				{
+					Key:   "key2",
+					Value: "value2",
+				},
+			},
+
+			LifecycleStage: models.LifecycleStageActive,
+		},
+		{
+			Name: "Test Experiment 2",
+			Tags: []models.ExperimentTag{
+				{
+					Key:   "key1",
+					Value: "value1",
+				},
+			},
+			LifecycleStage: models.LifecycleStageActive,
+		},
+		{
+			Name: "Test Experiment 3",
+			Tags: []models.ExperimentTag{
+				{
+					Key:   "key3",
+					Value: "value3",
+				},
+			},
+			LifecycleStage: models.LifecycleStageActive,
+		},
+		{
+			Name: "Test Experiment 4",
+			Tags: []models.ExperimentTag{
+				{
+					Key:   "key4",
+					Value: "value4",
+				},
+			},
+			LifecycleStage: models.LifecycleStageActive,
+		},
+		{
+			Name:           "Test Experiment 5",
+			Tags:           nil,
+			LifecycleStage: models.LifecycleStageActive,
+		},
+		{
+			Name:           "Test Experiment 6",
+			Tags:           nil,
+			LifecycleStage: models.LifecycleStageDeleted,
+		},
+	}
+	for _, ex := range experiments {
+		_, err := s.fixtures.CreateTestExperiment(context.Background(), &models.Experiment{
+			Name: ex.Name,
+			Tags: ex.Tags,
+			CreationTime: sql.NullInt64{
+				Int64: time.Now().UTC().UnixMilli(),
+				Valid: true,
+			},
+			LastUpdateTime: sql.NullInt64{
+				Int64: time.Now().UTC().UnixMilli(),
+				Valid: true,
+			},
+			LifecycleStage:   ex.LifecycleStage,
+			ArtifactLocation: "/artifact/location",
+		})
+		assert.Nil(s.T(), err)
+	}
+	defer func() {
+		assert.Nil(s.T(), s.fixtures.UnloadFixtures())
+	}()
+
+	tests := []struct {
+		name     string
+		request  request.SearchExperimentsRequest
+		expected []string
+	}{
+		{
+			name: "Test Filter",
```

Comment:
Please do not use `whitespaces` in test names. They will be changed to `underscores` so you won't find them in case if something just failed.